pkg/aws/client.go

```diff
@@ -0,0 +1,63 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+)
+
+// Client provides the single API client to make operations call to aws services
+type Client interface {
```

Comment:
From what I can tell, this interface is superfluous. It looks like we've defined an interface in a provider which is, in the majority of cases, unnecessary - let consumers define the interface with behaviors they expect.